src/olympia/devhub/views.py

```diff
@@ -1363,6 +1364,7 @@ def submit_version_agreement(request, addon_id, addon):
 
 
 @transaction.atomic
+@require_submissions_enabled
```

Comment:
Not necessary to protect this page (see comment below about needing to protect `upload()` and how to integrate the error message better in the submit upload page, though)